### ISSUE TITLE
Mono: Export templates won't build off master

### ISSUE THREAD

**Issue by exts** [godotengine/godot#22751]
**Godot version:**
Master

**OS/device including version:**
Windows 10

**Issue description:**
Can't build release export templates 

**Steps to reproduce:**
1. open x64 Native Tools Command Prompt VS 2017
2. run `scons p=windows target=release tools=no module_mono_enabled=yes -j3`
3. Expect error:

```
modules\mono\mono_gd\gd_mono.cpp(266): error C2065: 'editor_api_assembly': undeclared identifier
scons: *** [modules\mono\mono_gd\gd_mono.windows.opt.64.obj] Error 2
scons: building terminated because of errors.
```
https://github.com/godotengine/godot/blob/master/modules/mono/mono_gd/gd_mono.cpp#L266

**Minimal reproduction project:**
N/A

**Comment by exts** [comment]
I guess this is fixed with #22688

**Comment by exts** [comment]
New error after building master with #22688

```
d:\godot-devel\build\godot\modules\mono\mono_gd/gd_mono.h(189): error C2065: 'mono_reg_info': undeclared identifier
d:\godot-devel\build\godot\modules\mono\mono_gd/gd_mono.h(189): error C2065: 'mono_reg_info': undeclared identifier
scons: *** [modules\mono\mono_gc_handle.windows.opt.64.obj] Error 2
modules\mono\csharp_script.cpp(575): warning C4099: 'CSharpScriptDepSort': type name first seen using 'class' now seen using 'struct'
modules\mono\csharp_script.cpp(575): note: see declaration of 'CSharpScriptDepSort'
modules\mono\csharp_script.cpp(1197): warning C4805: '==': unsafe mix of type 'MonoBoolean' and type 'bool' in operation
d:\godot-devel\build\godot\modules\mono\mono_gd/gd_mono.h(189): error C2065: 'mono_reg_info': undeclared identifier
scons: *** [modules\mono\register_types.windows.opt.64.obj] Error 2
scons: *** [modules\mono\csharp_script.windows.opt.64.obj] Error 2
scons: building terminated because of errors.
```

cc @neikeq
